packages/http-server-fastify/src/FastifyHttpServer.ts

```diff
@@ -0,0 +1,216 @@
+/*
+ * © Copyright 2022 HP Development Company, L.P.
+ * SPDX-License-Identifier: MIT
+ */
+import {
+	ErrorRequestHandler,
+	HttpServerModule,
+	HttpServerModuleOptions,
+	ParameterSource,
+	RequestHandler
+} from '@davinci/http-server';
+import {
+	fastify,
+	FastifyInstance,
+	FastifyPluginCallback,
+	FastifyPluginOptions,
+	FastifyReply,
+	FastifyRequest
+} from 'fastify';
+import { Server as HttpServer } from 'http';
+import { Server as HttpsServer } from 'https';
+import type { App } from '@davinci/core';
+import fastifyCors, { FastifyCorsOptions } from '@fastify/cors';
+
+type Server = HttpServer | HttpsServer;
+
+type FastifyHttpServerModuleOptions = {
+	app?: FastifyInstance;
+	middlewares?: {
+		cors?: FastifyCorsOptions;
+	};
+	plugins?: [FastifyPluginCallback, FastifyPluginOptions?][];
+} & HttpServerModuleOptions;
+
+export class FastifyHttpServer extends HttpServerModule<
```

Comment:
Do we need to differentiate between what we extend from HttpServerModule and what we need to implement using `implements + Interface`?

Reply:
the specific express/fastify/other modules will have to extend the abstract HttpServerModule class, and that will:
- force them to implement the abstract method
- allow them to use some methods via `super`

What do you mean by differentiate them? Using some specific name convention?

Reply:
Just wondering if they will be able to not extend and do their own base class, but I'm fine with this if the intention is to make everyone extend this base class.

Reply:
obviously they could, the HttpServerModule is meant just to provide the same interface for registering routes and bringing them with validation and interceptors functionalities